[Feature] remove mode auto because google home has only eco and heat or cool

KartoffelToby opened the pull request:
## Breaking change
- Thermostat Auto Mode is gone


## Proposed change
Google Home Smarthome Action din't accept Mode "Auto" so only Heat or Eco is useful.
I use deconz with Euronics Spirit Zigbee Thermostats. Without changes i can't control it over GA (Shows undefined mode)
In "Auto" eq "Eco" mode i can't control the target temperature over Google Assistent otherwise the service is useless for me, and i think for others too. So. the Mode on HA should be "HVAC_MODE_AUTO" and on Google Home "Heat"

## Type of change
- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [X] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

Created a working Docker image with this changes: kartoffeltoby/homeassistent:2021.2.3

frenck committed:
Bumped version to 2021.3.0b0

homeassistant committed:
[ci skip] Translation update

raman325 committed:
Fix zwave_js unique ID migration logic (#47031)

bdraco committed:
Ensure doorbird events are re-registered when changing options (#46860)

- Fixed the update listener not being unsubscribed

- DRY up some of the code

- Fix sync code being called in async

- Reduce executor jumps

balloob committed:
Remove deprecated credstash + keyring (#47033)

chemelli74 committed:
Fix missing Shelly external input (#47028)

* Add support for external input (Shelly 1/1pm add-on)

* Make external sensor naming consistent

* Fix case consistency

cyberjunky committed:
Bump python-garminconnect to 0.1.19 to fix broken api (#47020)

ludeeus committed:
Use dispatch instead of eventbus for supervisor events (#46986)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>
Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

balloob committed:
Bumped version to 2021.3.0b1

bramkragten committed:
Updated frontend to 20210225.0 (#47059)

bramkragten committed:
Revert CORS changes for my home assistant (#47064)

* Revert CORS changes for my home assistant

* Update test_init.py

* Update test_init.py

homeassistant committed:
[ci skip] Translation update

raman325 committed:
catch ValueError when unique ID update fails because its taken and remove the duplicate entity (#47072)

bdraco committed:
Ensure hue options show the defaults when the config options have not yet been saved (#47067)

balloob committed:
Bump Z-Wave JS Server Python to 0.20.0 (#47076)

balloob committed:
Bumped version to 2021.3.0b2

balloob committed:
Upgrade aiohttp to 3.7.4 (#47077)

emontnemery committed:
Bump pychromecast to 8.1.2 (#47085)

Commit:
Add support for Shelly SHBTN-2 device triggers (#46644)

rikroe committed:
Bump bimmer_connected to 0.7.15 and fix bugs (#47066)

Co-authored-by: rikroe <rikroe@users.noreply.github.com>

agners committed:
Add new machine generic-x86-64 to build matrix (#47095)

The Intel NUC machine runs on most UEFI capable x86-64 machines today.
Lets start a new machine generic-x86-64 which will replace intel-nuc
over time.

bdraco committed:
Add suggested area to hue (#47056)

elupus committed:
Add support for v6 features to philips js integration (#46422)

KartoffelToby committed:
[TASK] remove mode auto because google home has no

homeassistant commented:
Hi @KartoffelToby, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#47107).

Once you do that we will be able to review and accept this pull request.

Thanks!

chemelli74 committed:
Bump aioshelly to 0.6.1 (#47088)

balloob commented:
https://developers.google.com/assistant/smarthome/traits/temperaturesetting

Your statement is wrong. `auto` still very much is a mode in Google.

marcelveldt committed:
Fix Z-Wave JS discovery schema for thermostat devices (#47087)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

allenporter committed:
Bump google-nest-sdm to v0.2.12 to improve API call error messages (#47108)

balloob committed:
Bumped version to 2021.3.0b3

KartoffelToby commented:
@balloob Thats strange i got an "Undefined" Blackout Thermostat setting in GA if i run the current container without the mode change.

balloob committed:
Use async_capture_events to avoid running in executor (#47111)

thecode committed:
Fix Shelly RGBW (#47116)

homeassistant committed:
[ci skip] Translation update

bramkragten committed:
Update frontend to 20210226.0 (#47123)

bdraco committed:
Provide a human readable exception for the percentage util (#47121)

bdraco committed:
Handle lutron_caseta fan speed being none (#47120)

elupus commented:
We are providing it as comma separated list instead of array though. Maybe they don't accept it in the old format.

homeassistant committed:
[ci skip] Translation update

agners committed:
Bump builder to get generic-x86-64 nightly builds (#47164)

balloob committed:
Fix the updater schema (#47128)

dmulcahey committed:
Bump ZHA quirks to 0.0.54 (#47172)

bdraco committed:
Update HAP-python to 3.3.1 (#47180)

Fixes disconnect when setting a single char fails
https://github.com/ikalchev/HAP-python/compare/v3.3.0...v3.3.1

emontnemery committed:
Fix MQTT trigger where wanted payload may be parsed as an integer (#47162)

balloob committed:
Bumped version to 2021.3.0b4

starkillerOG committed:
Fix Xiaomi Miio discovery (#47134)

ajschmidt8 committed:
Update AlarmDecoder dependency (#46841)

homeassistant committed:
[ci skip] Translation update

bachya committed:
Bump simplisafe-python to 9.6.7 (#47206)

bdraco committed:
Increment the homekit config version when restarting (#47209)

If an entity changes between restart the iOS/controller
device may have cached the old chars for the accessory.
To force the iOS/controller to reload the chars, we
increment the config version when Home Assistant restarts

bdraco committed:
Bump HAP-python to 3.3.2 to fix unavailable condition on restart (#47213)

Fixes

https://github.com/ikalchev/HAP-python/compare/v3.3.1...v3.3.2

agners committed:
Fix generic-x86-64 build (#47214)

Replace the wrong Docker Hub repository slipped in during testing.

pvizeli committed:
Bumped version to 2021.3.0b5

maxwroc committed:
Fix number of reported issues by github integration (#47203)

emontnemery committed:
Fix race when disabling config entries (#47210)

* Fix race when disabling config entries

* Remove unused constant

starkillerOG committed:
Fix Xiaomi Miio flow unique_id for non discovery flows (#47222)

thecode committed:
Fix Shelly Polling (#47224)

emontnemery committed:
Pass variables to initial evaluation of template trigger (#47236)

* Pass variables to initial evaluation of template trigger

* Add test

* Clarify test

bdraco committed:
Fix lutron caseta fan handling of speed off (#47244)

raman325 committed:
Add suggested area for zwave_js devices (#47250)

bramkragten committed:
Update frontend to 20210301.0 (#47252)

bachya committed:
Bump simplisafe-python to 9.6.8 (#47241)

balloob committed:
Revert "Fix the updater schema (#47128)" (#47254)

This reverts commit 98be703d90e44efe43b1a17c7e5243e5097b00b1.

frenck committed:
Fix duplicate template handling in Persistent Notifications (#47217)

bdraco committed:
Fix harmony failing to switch activities when a switch is in progress (#47212)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

marcelveldt committed:
Update color logic for zwave_js light platform (#47110)

Co-authored-by: Raman Gupta <7243222+raman325@users.noreply.github.com>

balloob committed:
Bumped version to 2021.3.0b6

emontnemery committed:
Make MQTT number respect retain setting (#47270)

KartoffelToby commented:
Okay i discoverd a other problem with the current set Temperatures after modes are toggled that break the GA too.

i'll debug the current stable Version to find out whats going wrong.

balloob committed:
Fix Alexa doorbells (#47257)

bramkragten committed:
Update frontend to 20210302.0 (#47278)

bachya committed:
Bump simplisafe-python to 9.6.9 (#47273)

Adminiuga committed:
Update ZHA dependencies (#47282)

chemelli74 committed:
Limit log spam by ConfigEntryNotReady (#47201)

frenck committed:
Upgrade pillow to 8.1.1 (#47223)

cgtobi committed:
Fix issue when setting boost preset for a turned off Netatmo thermostat (#47275)

raman325 committed:
Add raw values to zwave_js value notification event (#47258)

* add value_raw to value notification event that always shows the untranslated state value

* add property key and property to event params

raman325 committed:
Convert climacell forecast timestamp to isoformat so that UI shows the right times (#47286)

raman325 committed:
Bump zwave-js-server-python to 0.20.1 (#47289)

MartinHjelmare committed:
Add zwave_js add-on manager (#47251)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

balloob committed:
Bumped version to 2021.3.0b7

raman325 committed:
Correct climacell device info (#47292)

bramkragten committed:
Update frontend to 20210302.3 (#47310)

emontnemery committed:
Improve behaviour when disabling or enabling config entries (#47301)

balloob committed:
Simplify switch light (#47317)

frenck committed:
Bumped version to 2021.3.0

frenck committed:
Merge pull request #47319 from home-assistant/rc

elupus commented:
Do you think you could update the code provide a proper list instead of a comma separated set of values as it does now in the sync response? It might solve your issue.

balloob commented:
Why is your device reporting AUTO mode to begin with and not MODE_HEAT_COOL?

amelchio committed:
Catch ConditionError in generic_thermostat climate (#47359)

starkillerOG committed:
Fix Xiaomi Miio setup of switch entity for lumi.acpartner.v3 (#47345)

bdraco committed:
Map silent as a preset mode for fan backcompat (#47396)

The original change did not map silent as a preset mode
because it was not clear if it was a speed or a preset.

esev committed:
Revert "Speed-up wemo discovery (#46821)" (#47392)

This reverts commit 6e52b26c06098052d379065b00f570c2a44653e1.

frenck committed:
Don't raise on known non-matching states in numeric state condition (#47378)

bramkragten committed:
Update frontend to 20210302.4 (#47383)

MartinHjelmare committed:
Fix zwave_js manual reconfiguration of add-on managed entry (#47364)

MartinHjelmare committed:
Fix access of missing zwave_js climate unit value (#47380)

syssi committed:
Fix measurement unit (Closes: #47390) (#47398)

raman325 committed:
Don't convert Climacell forecast temperatures to celsius because platform does it automatically (#47406)

bachya committed:
Fix AirVisual exception when config entry contains old integration type (#47405)

raman325 committed:
Fix Climacell timezone issue with daily forecasts (#47402)

frenck committed:
Fix older Roborock models (#47412)

MartinHjelmare committed:
Make zwave_js add-on manager more flexible (#47356)

cpainchaud committed:
Fix RFLink TCP KeepAlive error log (#47395)

raman325 committed:
Bump zwave-js-server-python to 0.21.0 (#47408)

Co-authored-by: Tobias Sauerwein <cgtobi@users.noreply.github.com>

balloob committed:
Bumped version to 2021.3.1

MartinHjelmare committed:
Only create snapshot if add-on update will be done (#47424)

balloob committed:
Merge pull request #47422 from home-assistant/rc

raman325 committed:
Bump zwave-js-server-python to 0.21.1 (#47464)

bramkragten committed:
Update frontend to 20210302.5 (#47462)

elupus committed:
Bump version with fix for v1 (#47458)

frenck committed:
Fix Hue scene overriding Hue default transition times (#47454)

functionpointer committed:
Use conn_made callback in MySensors (#47463)

mvn23 committed:
Update pyotgw to 1.1b1 (#47446)

raman325 committed:
Update zwave_js.refresh_value service description (#47469)

cgtobi committed:
Fix issue at Netatmo startup (#47452)

pnbruckner committed:
Bump amcrest package version to 1.7.1 (#47483)

balloob committed:
Raise error instead of crashing when template passed to call service target (#47467)

balloob committed:
Bumped version to 2021.3.2

balloob committed:
Merge pull request #47490 from home-assistant/rc

amelchio committed:
Fix Sonos polling mode (#47498)

natekspencer committed:
Adjust litterrobot tests and code to match guidelines (#47060)

* Use SwitchEntity instead of ToggleEntity and adjust test patches as recommended

* Move async_create_entry out of try block in config_flow

* Patch pypi package instead of HA code

* Bump pylitterbot to 2021.2.6, fix tests, and implement other code review suggestions

* Bump pylitterbot to 2021.2.8, remove sleep mode start/end time from vacuum, adjust and add sensors for sleep mode start/end time

* Move icon helper back to Litter-Robot component and isoformat times on time sensors

MartinHjelmare committed:
Fix mysensors notify platform (#47517)

MartinHjelmare committed:
Bump pymysensors to 0.21.0 (#47530)

MartinHjelmare committed:
Fix mysensors device tracker (#47536)

bdraco committed:
Bump HAP-python to 3.4.0 (#47476)

* Bump HAP-python to 3.3.3

* bump

* fix mocking

MartinHjelmare committed:
Fix mysensors unload clean up (#47541)

dgomes committed:
Correct weather entities forecast time (#47565)

bramkragten committed:
Update frontend to 20210302.6 (#47592)

raman325 committed:
Add fallback zwave_js entity name using node ID (#47582)

* add fallback zwave_js entity name using node ID

* add new fixture and test for name that was failing

thecode committed:
Fix Shelly logbook exception when missing COAP (#47620)

bdraco committed:
Ensure template fan value_template always determines on state (#47598)

ronytomen committed:
Allow running and restarting with both ozw and zwave active (#47566)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

ollo69 committed:
Fix AsusWRT wrong api call (#47522)

bdraco committed:
Fix turn on without speed in homekit controller (#47597)

bdraco committed:
Ensure bond devices recover when wifi disconnects and reconnects (#47591)

bdraco committed:
Fix turning off scene in homekit (#47604)

bdraco committed:
Fix insteon fan speeds (#47603)

raman325 committed:
Update zwave_js supported features list to be static (#47623)

balloob committed:
Bumped version to 2021.3.3

balloob committed:
Merge pull request #47645 from home-assistant/rc

KartoffelToby committed:
Merge tag '2021.3.3' into fix-google-home-thermostat-mode

elupus commented:
This ended up a broken rebase. Lots of unrelated commits. Please create a new pull for any followup.
